binding/android/Cobra/cobra/src/main/java/ai/picovoice/cobra/Cobra.java

```diff
@@ -107,4 +116,71 @@ public String getVersion() {
         return CobraNative.getVersion();
     }
 
+    /**
+     * Lists all available devices that Cobra can use for inference.
+     * Each entry in the list can be used as the `device` argument when initializing Cobra.
+     *
+     * @return Array of all available devices that Cobra can be used for inference.
+     * @throws CobraException if getting available devices fails.
+     */
+    public static String[] getAvailableDevices() throws CobraException {
+        return CobraNative.listHardwareDevices();
+    }
+
+    /**
+     * Builder for creating an instance of Cobra with a mixture of default arguments.
+     */
+    public static class Builder {
+
+        private String accessKey = null;
+        private String device = "best";
```

Comment:
default value should be null and then you'll swap in the `build()` function with :
```java
if (device == null) {
    device = "best";
}
```